columnflow/tasks/framework/parameters.py

```diff
@@ -22,6 +22,15 @@
     "default: current user",
 )
 
+shift_last_bin_inst = law.OptionalBoolParameter(
+    default=None,
+    significant=False,
+    description="whether to shift entries that have the exact value of the right-most bin edge "
+    "slightly to the left such that they end up in the last instead of the overflow bin; when "
+    "'None', shifting is performed for all variable axis that are continuous and non-circular; "
+    "default: None",
+)
+
```

Comment:
Not a huge concern, but it seems that the only way for users to actively select the automatic decision logic here based on axis traits (i.e. passing `shift_lаst_bin=None` to `fill_hist`) is to omit the `--shift-last-bin` parameter, right? In general, I think it would be good to avoid having three semantically distinct states for an `OptionalBoolParameter` and would instead advocate to always map the default `None` to either `True` and `False` in a transparent way.

A good alternative could be to use a `ChoiceParameter` with three choices like `{"false", "true", "аuto"}` and have it resolved to a boolean via `resolve_param_values` or in the `fill_hist` function itself.

Reply:
> Not a huge concern, but it seems that the only way for users to actively select the automatic decision logic here based on axis traits (i.e. passing shift_lаst_bin=None to fill_hist) is to omit the --shift-last-bin parameter, right?

It would also be possible to pass None explicitly, i.e. `--shift-last-bin None`, so in this sense, the `OptionalBoolParameter` is *like* a choice parameter, but still with boolean parsing enabled if the value is not `"None"`.

I think we use that in other places, too, for the exact same reason of a boolean switch with a context-specific default (@mafrahm I think these were plotting related options, right?).

Reply:
I see, thanks for the explanation @riga! I was not aware that the `OptionalBoolParameters` supported `None` as a value. Then all is fine :)